apteryx.c

```diff
@@ -693,6 +693,18 @@ apteryx_get_string (const char *path, const char *key)
     return str;
 }
 
+char *
+apteryx_get_string_default (const char *path, const char *key, char *deflt)
+{
+    char *value = NULL;
+    value = apteryx_get_string (path, key);
+    if (!value)
+    {
+        return deflt;
```

Comment:
You'll want to pass in a const char *deflt and duplicate it here - it'll make memory management a whole lot easier for callers of this function.